### PR TITLE
Update front end to display multiple query results.

### PR DESCRIPTION
**Pull request by saakshisaxena**
Update front end to display multiple query results.

Create a class for storing a query and its shopping results (products) and send it in the form of Json to JavaScript.
From JavaScript display the list of queries we formed, ie through detection of image, followed by blocks of query and its results.

### CODE REVIEW

File: src/main/webapp/style.css [hunk 1]
```diff
@@ -120,3 +120,10 @@ body {
   position: absolute;
   width: 100%;
 }
+
+.query {
+  font-weight: bold;
+  font-size: large;
```
Comment (ralucatudor) [inline src/main/webapp/style.css:126]: Nit: switch lines 125 and 126 for alphabetical order

File: src/main/webapp/script.js [hunk 2]
```diff
@@ -64,69 +64,78 @@ async function onSubmitUploadImageForm() {
...
-  // Empty the prompt container and add the {@code products} content into the page.
+  // Empty the prompt container and add the query and its resultant content into the page.
   $('#search-loading-prompt').empty();
   $('#loading-gif-prompt').addClass('hidden');
 
-  // Show the user the shopping query built to search on Google Shopping.
   $('#shopping-query-display-container').removeClass('hidden');
-  $('#shopping-query').text(shoppingQuery);
+  // So that the queries list starts at a new line
+  $('#shopping-query').append('<br />');
```
Comment (ralucatudor) [inline src/main/webapp/script.js:91]:
```suggestion
  // Add a new line before displaying the queries.
  $('#shopping-query').append('<br />');
```

File: src/main/webapp/script.js [hunk 3]
```diff
@@ -64,69 +64,78 @@ async function onSubmitUploadImageForm() {
...
-
-    // Load the content using jQuery's append.
-    $productContainer.append(productElementHTML);
-
-    // Add the container to the results page, into the corresponding product wrapper.
-    $('#shopping-results-wrapper').append($productContainer);
-  });
+async function appendShoppingResults(shoppingResults) {
+  shoppingResults.forEach((result) => {
+          // Show the user the shopping query built to search on Google Shopping.
```
Comment (ralucatudor) [inline src/main/webapp/script.js:101]: Nit: let's use 4 spaces for this forEach block (here I think are 8)
```suggestion
  shoppingResults.forEach((result) => {
      // Show the user the shopping query built to search on Google Shopping.
```
and so for the next lines

File: src/main/webapp/script.js [hunk 4]
```diff
@@ -64,69 +64,78 @@ async function onSubmitUploadImageForm() {
...
+          $('#shopping-query').append(result['query'] + '<br />');
+          const $query = `<div class = 'text-center query'>${result['query']}</div>`;
```
Comment (ralucatudor) [inline src/main/webapp/script.js:103]:
```suggestion
          const $query = `<div class='text-center query'>${result['query']}</div>`;
```

File: src/main/java/com/google/sps/servlets/HandlePhotoShoppingServlet.java [hunk 5]
```diff
@@ -98,27 +99,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
         try {
-            shoppingQuerierResults.add(querier.query(shoppingQueryInput));
+        shoppingQuerierResults = querier.query(shoppingQueryInput);
         } catch (IllegalArgumentException
             | ShoppingQuerierConnectionException
             | IOException exception) {
             response.sendError(SC_INTERNAL_SERVER_ERROR, exception.getMessage());
             return;
         }
+    shoppingResults.add(ShoppingResult.create(shoppingQueryInput.getShoppingQuery(), shoppingQuerierResults));
```
Comment (ralucatudor) [inline src/main/java/com/google/sps/servlets/HandlePhotoShoppingServlet.java:113]: Tiny nit:
```suggestion
      shoppingResults.add(ShoppingResult.create(shoppingQueryInput.getShoppingQuery(), shoppingQuerierResults));
```

File: src/main/webapp/script.js [hunk 6]
```diff
@@ -64,69 +64,78 @@ async function onSubmitUploadImageForm() {
...
+          $('#shopping-results-wrapper').append($query);
```
Comment (ralucatudor) [inline src/main/webapp/script.js:104]: I might not get this so please help me out - here is the query displayed two times on the page?

**Review by ralucatudor — COMMENTED** [review]
I like the idea with creating the ShoppingResult class!! 👍

File: src/main/java/com/google/sps/servlets/HandlePhotoShoppingServlet.java [hunk 7]
```diff
@@ -98,27 +99,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
     GoogleShoppingQuerier querier = new GoogleShoppingQuerier();
 
-    List<List<Product>> shoppingQuerierResults = new ArrayList<>();
+    List<ShoppingResult> shoppingResults = new ArrayList<>();
+    List<Product> shoppingQuerierResults = new ArrayList<>();
     for(ShoppingQueryInput shoppingQueryInput : shoppingQueryInputs) {
         try {
-            shoppingQuerierResults.add(querier.query(shoppingQueryInput));
+        shoppingQuerierResults = querier.query(shoppingQueryInput);
         } catch (IllegalArgumentException
```
[the remark this answers is not in the document]
Comment (saakshisaxena) [reply src/main/java/com/google/sps/servlets/HandlePhotoShoppingServlet.java:107]: I'll run it through the linter again :) and if it still does that then I'll change it manually.

File: src/main/webapp/script.js (hunk 3 above)
Comment (saakshisaxena) [reply src/main/webapp/script.js:101]: I'll run it through the linter once again

File: src/main/webapp/script.js (hunk 6 above)
Comment (saakshisaxena) [reply src/main/webapp/script.js]: So here I'm displaying all the queries that we are showing results for and then displaying query-results blocks, to make sure and to let the user know that its not just one product they are searching for.

I just deployed it again you should be able to test it.

Comment (saakshisaxena) [reply src/main/webapp/script.js]: In another pr we could make those as links to that particular block of results.

**Review by ralucatudor — APPROVED** [review]

File: src/main/webapp/style.css [hunk 8]
```diff
@@ -120,3 +120,10 @@ body {
   position: absolute;
   width: 100%;
 }
+
+.query {
+  font-size: large;
+  font-weight: bold;
+  margin: 3%;
+  padding: 3%;
```
Comment (skopecki) [inline src/main/webapp/style.css:128]: I would use fixed values instead of %.

File: src/test/java/com/google/sps/ImageTextDectectorTest.java [hunk 9]
```diff
@@ -91,6 +91,7 @@ public void noText() throws Exception {
   public void singleWordImage() throws Exception {
```
Comment (skopecki) [inline src/test/java/com/google/sps/ImageTextDectectorTest.java:91]: Changes to this test are unrelated to this change. Please remove them.

File: src/main/webapp/script.js [hunk 10]
```diff
@@ -65,69 +65,78 @@ async function onSubmitUploadImageForm() {
...
-                                                     product.priceAndSeller,
-                                                     product.link,
-                                                     product.shippingPrice);
-
-    // Load the content using jQuery's append.
-    $productContainer.append(productElementHTML);
-
-    // Add the container to the results page, into the corresponding product wrapper.
-    $('#shopping-results-wrapper').append($productContainer);
+async function appendShoppingResults(shoppingResults) {
```
Comment (skopecki) [inline src/main/webapp/script.js:100]: This function is not `async`.

File: src/main/webapp/script.js [hunk 11]
```diff
@@ -65,69 +65,78 @@ async function onSubmitUploadImageForm() {
...
-  // Empty the prompt container and add the {@code products} content into the page.
+  // Empty the prompt container and add the query and its resultant content into the page.
   $('#search-loading-prompt').empty();
   $('#loading-gif-prompt').addClass('hidden');
 
-  // Show the user the shopping query built to search on Google Shopping.
   $('#shopping-query-display-container').removeClass('hidden');
-  $('#shopping-query').text(shoppingQuery);
+  // Add a new line before displaying the queries.
+  $('#shopping-query').append('<br />');
```
Comment (skopecki) [inline src/main/webapp/script.js:92]: Remove this from here.

File: src/main/webapp/script.js [hunk 12]
```diff
@@ -65,69 +65,78 @@ async function onSubmitUploadImageForm() {
...
-
-    // Load the content using jQuery's append.
-    $productContainer.append(productElementHTML);
-
-    // Add the container to the results page, into the corresponding product wrapper.
-    $('#shopping-results-wrapper').append($productContainer);
+async function appendShoppingResults(shoppingResults) {
+  shoppingResults.forEach((result) => {
+    // Show the user the shopping query built to search on Google Shopping.
+    $('#shopping-query').append(result['query'] + '<br />');
```
Comment (skopecki) [inline src/main/webapp/script.js:103]: Swap this around: `'<br>' + result['query']`. Also use `<br>`.

File: src/main/webapp/script.js [hunk 13]
```diff
@@ -65,69 +65,78 @@ async function onSubmitUploadImageForm() {
...
+    const $query = `<div class='text-center query'>${result['query']}</div>`;
+    $('#shopping-results-wrapper').append($query);
+
+    const $listProductsContainer = $('<div>', {class: 'row'});
+
+    const $products = result['products'];
```
Comment (skopecki) [inline src/main/webapp/script.js:109]: Don't use `$` in your variable names.

File: src/main/webapp/script.js (hunk 10 above)
Comment (ralucatudor) [reply src/main/webapp/script.js:100]: Oops, this is my fault, I must have copied a function header when first writing this function and, by mistake, left async there, sorry 👍 Good catch, I hadn't seen it!

File: src/test/java/com/google/sps/ImageTextDectectorTest.java (hunk 9 above)
Comment (saakshisaxena) [reply src/test/java/com/google/sps/ImageTextDectectorTest.java:91]: Hmm, so here I realised that the test cases I was taking previously were missing the first line returned by Cloud Vision API ( I checked it by logging the results from API), so I decided to add those here.

Are you suggesting I make a new pr for these changes?

File: src/main/webapp/script.js (hunk 13 above)
Comment (saakshisaxena) [reply src/main/webapp/script.js:109]: So, should I not use `$` for any variable in the whole .js file?

File: src/test/java/com/google/sps/ImageTextDectectorTest.java (hunk 9 above)
Comment (skopecki) [reply src/test/java/com/google/sps/ImageTextDectectorTest.java:91]: Yes, please remove it from this PR. It has nothing to do with this PR and it looks like something which got added by accident.

I did comment on this change in the other PR.

File: src/main/webapp/script.js (hunk 13 above)
Comment (skopecki) [reply src/main/webapp/script.js:109]: Google's style guide says: "Identifiers use only ASCII letters and digits, and, in a small number of cases noted below, underscores and very rarely (when required by frameworks like Angular) dollar signs."
https://google.github.io/styleguide/jsguide.html#naming-rules-common-to-all-identifiers

I know there are other conventions out there, but it's good to stick to one style guide. If you're following another style guide and you use the dollar signs consistently that's also fine.

File: src/main/webapp/style.css (hunk 8 above)
Comment (saakshisaxena) [reply src/main/webapp/style.css:128]: I was thinking if I used fixed values then it might not work well on mobile phones and smaller devices

Comment (skopecki) [reply src/main/webapp/style.css:128]: SGTM

**Review by skopecki — APPROVED** [review]
